xarray/test/test_dataarray.py

```diff
@@ -1210,6 +1213,106 @@ def test_groupby_first_and_last(self):
+
+        self.assertEqual(rolling_obj._axis_num, 0)
+
+        # catching invalid args
+        with self.assertRaisesRegexp(ValueError, 'only one dim/window can be'):
+            da.rolling(time=7, x=2)
+        with self.assertRaisesRegexp(ValueError, 'window must be > 0'):
+            da.rolling(time=-2)
+
+    @requires_bottleneck
+    def test_rolling_wrapped_bottleneck(self):
+        import bottleneck as bn
+
+        da = self.make_rolling_example_array()
+
+        # Test all bottleneck functions
+        rolling_obj = da.rolling(time=7)
+        for name in ('sum', 'mean', 'std', 'min', 'max', 'median'):
+            func_name = 'move_{0}'.format(name)
+            actual = getattr(rolling_obj, name)()
+            expected = getattr(bn, func_name)(da.values, window=7, axis=0)
+            self.assertArrayEqual(actual.values, expected)
+
+        # Using min_periods
+        rolling_obj = da.rolling(time=7, min_periods=1)
+        for name in ('sum', 'mean', 'std', 'min', 'max'):
+            func_name = 'move_{0}'.format(name)
+            actual = getattr(rolling_obj, name)()
+            expected = getattr(bn, func_name)(da.values, window=7, axis=0,
+                                              min_count=1)
+            self.assertArrayEqual(actual.values, expected)
+
+        # Using center=False
+        rolling_obj = da.rolling(time=7, center=False)
+        for name in ('sum', 'mean', 'std', 'min', 'max', 'median'):
+            actual = getattr(rolling_obj, name)()['time']
+            self.assertDataArrayEqual(actual, da['time'])
+
+        # Using center=True
+        rolling_obj = da.rolling(time=7, center=True)
+        for name in ('sum', 'mean', 'std', 'min', 'max', 'median'):
+            actual = getattr(rolling_obj, name)()['time']
+            self.assertDataArrayEqual(actual, da['time'])
+
+        # catching invalid args
+        with self.assertRaisesRegexp(ValueError, 'Rolling.median does not'):
+            da.rolling(time=7, min_periods=1).median()
+
+    def test_rolling_pandas_compat(self):
+        s = pd.Series(range(10))
+        da = DataArray.from_series(s)
+
+        for center in (False, True):
+            for window in [1, 2, 3, 4]:
+                for min_periods in [None, 0, 1, 2]:
+                    if min_periods is not None and window < min_periods:
+                        min_periods = window
+                    s_rolling = pd.rolling_mean(s, window, center=center,
+                                                min_periods=min_periods)
+                    da_rolling = da.rolling(index=window, center=center,
+                                            min_periods=min_periods).mean()
+                    # pandas does some fancy stuff in the last position
+                    np.testing.assert_allclose(s_rolling.values[:-1],
+                                               da_rolling.values[:-1])
+                    np.testing.assert_allclose(s_rolling.index,
```

Comment:
> @shoyer - The only inconsistency I can find between my approach here and how Pandas does this is when center == True and window >= 3'. The difference only occurs in the last position. We are using the shift method to adjust for center but it seems pandas treats the last position differently in this case. I'm not quite sure how to proceed here.

This is still the only failing test in this PR.  I'd like to consider accepting the work done here with the caveat that xarray rolling differs from pandas in this one way. A possible followup PR may be able to reconcile this difference.